deep_qa/data/instances/sequence_tagging/verb_semantics_instance.py

```diff
@@ -0,0 +1,173 @@
+        words['tags'] = self.label[1]
+        return words
+
+    @overrides
+    def to_indexed_instance(self, data_indexer: DataIndexer):
+        sentence_indices = self._index_text(self.sentence, data_indexer)
+        verb_indices = sentence_indices[self.verb[0]:self.verb[1]]
+        entity_indices = sentence_indices[self.entity[0]:self.entity[1]]
+
+        indexed_label = self._index_label()
+        return IndexedVerbSemanticsInstance(sentence_indices, verb_indices,
+                                            entity_indices, indexed_label, self.index)
+
+    def _index_label(self, data_indexer: DataIndexer) -> (int, List[int]):
+        """ 
+        indexed_label = [state_change, tags]
+        state_change = [1-hot vector of size #change_types]
+        tags = [list of size sentence_length [1-hot vector of size #tag_types] ]
+        """
+
+        state_index = data_indexer.get_word_index(self.label[0], namespace='state_changes')
+        state_one_hot = numpy.zeros(self.data_indexer.get_vocab_size(namespace='state_changes') - 2)
+        state_one_hot[self.label[0] - 2] = 1
+
+        tag_one_hot_list = []
+        tag_indices = [data_indexer.get_word_index(tag, namespace='tags') for tag in self.label[1]]
+        for tag_index in tag_indices:
+            # We subtract 2 here to account for the unknown and padding tokens that the DataIndexer
+            # uses.
+            tag_one_hot = numpy.zeros(data_indexer.get_vocab_size(namespace='tags') - 2)
+            tag_one_hot[tag_index - 2] = 1
+            tag_one_hot_list.append(tag_one_hot)
+
+        return (state_one_hot, tag_one_hot_list)
+
+    @classmethod
+    @overrides
+    def read_from_line(cls, line: str):
+        """
+        Reads a VerbSemanticsInstance object from a line.  The format is as follows:
+        [sentence][tab][verb][tab][entity][tab][stateChangeLabel][tab][arg1][tab][arg2]
+
+        sentence: this is pre-tokenized with tokens seperated by "####"
+        verb: startIndex,endIndex of verb in the sentence
+        entity: startIndex,endIndex of entity in the sentence
+        state_change_label: one label from {CREATE, DESTROY, MOVE}
+        arg1: startIndex,endIndex of arg1 in the sentence
+        arg2: startIndex,endIndex of arg2 in the sentence
+        """
+        fields = line.split("\t")
+
+        if len(fields) == 6:
+            sentenceStr, verb_str, entity_str, state_change_label, arg1_str, arg2_str = fields
+            sentence = sentenceStr.split("####")
+            v_parts = verb_str.split(",")
+            verb = (int(v_parts[0]), int(v_parts[1]))
+            e_parts = entity_str.split(",")
+            entity = (int(e_parts[0]), int(e_parts[1]))
+            a1_parts = arg1_str.split(",")
+            arg1 = (int(a1_parts[0]), int(a1_parts[1]))
+            a2_parts = arg2_str.split(",")
+            arg2 = (int(a2_parts[0]), int(a2_parts[1]))
+            return cls(sentence, verb, entity, state_change_label, arg1, arg2)
```

Comment:
It looks like this logic will crash if `arg1_str` or `arg2_str` is empty, which I'm pretty sure you expect to happen.